sigma/backends/loki/loki.py

```diff
@@ -1039,12 +1109,43 @@ def finalize_query(
         elif query is None:
             query = ""
         if state.has_deferred():
-            query = self.deferred_separator.join(
-                (
-                    deferred_expression.finalize_expression()
-                    for deferred_expression in state.deferred
+            standard_deferred = [
+                expression.finalize_expression()
+                for expression in state.deferred
+                if not isinstance(
+                    expression,
+                    (
+                        LogQLDeferredFieldRefExpression,
+                        LogQLDeferredFieldRefFilterExpression,
+                    ),
+                )
+            ]
+            field_refs = [
+                expression.finalize_expression()
+                for expression in state.deferred
+                if isinstance(expression, LogQLDeferredFieldRefExpression)
+            ]
+            field_ref_filters = [
+                expression.finalize_expression()
+                for expression in state.deferred
+                if isinstance(expression, LogQLDeferredFieldRefFilterExpression)
+            ]
+            field_ref_expression = ""
+            field_ref_filters_expression = ""
+            if len(field_refs) > 0:
+                label_fmt = ",".join(field_refs)
+                field_ref_expression = f"{'' if len(query) > 0 else '| ' + str(self.select_log_parser(rule)) + ' '}| label_format {label_fmt}"

```

Comment:
I don't think this makes sense to be an f-string - the majority of the line is a condition with the relevant value appended at the end - I personally try to avoid logic in them

Reply:
Also addressed in 8aed4ef